docs/reference/contributing/guidelines/workflow.md

```diff
@@ -79,6 +79,43 @@ Each feature has a tech lead. This person is responsible for:
 - Rebasing often to track master development.
 - Reviewing any addition to the feature branch (approval required by the feature tech lead or another assigned person).
 
+##### Pull request types
+
+We consider the following pull request types.
+
+##### Fix
+
+A bug fix should be backward-compatible internal change that fixes incorrect behavior. The fix should provide a description what and how it fixes the issue. The test should be included to catch this issue in the future and confirm the correct behavior.
+
+Release: patch version
+
+##### Refactor
+
+Refactors are intended for minor versions if they are not fixing specific issue as they 
+can introduce new issues.
+
+Release: minor version
+
+##### New target
+
+Adding a new target targets the patch release as it updating targets folder implementation.
+
+Release: patch version
+
+##### Feature
+
+New features targets minor releases. It can be integrated only if the feature supports most of the targets (if it requires new target HAL implementation).
+
+Adding a new functionality is considered to be a feature. It does not matter if it is C++ or C.
+
+Release: minor version
+
+##### Breaking change
+
+Any change that results in breaking user space. It should have strong justification to be considered. Often the change could be made backward compatible, for example deprecate the old functionality and introduce the new replacement.
+
+Release: minor version
```

Comment:
Shouldn't this be major version?

Reply:
Nope, major version is the 5. , next one will be 6. in a year's time

Reply:
major would be 6.x , we do 5.x - minor.. .sometimes called feature release. 

@AnotherButler what shall we define here ?

Reply:
But we don't making breaking changes on minor releases.

(If we do those should be painful exceptions)

Looking at the current rules, what's the motivation for a contributor to make a change "feature" vs "breaking"? They both get released at the same time.

Reply:
It is just a classification. If someone rewrites an algorithm which totally changes the behaviour of the system, then that is a breaking change but not a new feature....

Reply:
> But we don't make breaking changes on minor releases.

Yes, this should not be seen often and thus should really provide strong reason to be considered. there are usually ways to do it backward compatible, but might not always be the case.

> It is just a classification. If someone rewrites an algorithm which totally changes the behaviour of the system, then that is a breaking change but not a new feature....

Proposals for better specification of a breaking change ? It can be fix, feature or refactor if it results in breaking a user space. we can provide better examples here?

Reply:
A breaking changes is any change, other than the introduction of a new feature, which changes the user experience of the functionality.